### PR TITLE
[DOCS] GX Cloud Content Adjustments

### PR DESCRIPTION
**Pull request by kwcanuck**
In [this Slack conversation](https://greatexpectationslabs.slack.com/archives/C06TBEX44VC/p1712875461384989) @Shinnnyshinshin and @annabarr discussed the consolidation and edit of the content in the _[Self-hosted and org-hosted deployments](https://docs.greatexpectations.io/docs/cloud/deploy_gx_agent/#self-hosted-and-org-hosted-deployments)_ . In [a separate Slack conversation](https://greatexpectationslabs.slack.com/archives/CBWF8GCDT/p1712862976026699?thread_ts=1712862315.169609&cid=CBWF8GCDT), there was a request to update the annotation in _[Connect to GX Cloud with Python](https://docs.greatexpectations.io/docs/cloud/connect/connect_python)_ to clarify what should be done when a user doesn't want to deploy the GX Agent. This PR implements the requested changes.

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] Appropriate tests and docs have been updated

### CODE REVIEW

**Review by Rachel-Reverie — APPROVED** [review]
LGTM

File: docs/docusaurus/docs/cloud/deploy_gx_agent.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ import Tabs from '@theme/Tabs';
...
 
-The GX Agent is typically deployed in your organization's deployment environment, for example, in a development, staging, or production cloud services environment. The GX Agent serves all GX Cloud users within your organization. It can be run as part of your development or production workflows. To learn more about the GX Agent and deployment patterns, see [About GX Cloud](./about_gx.md).
+To try GX Cloud, you use a [self-hosted deployment](./about_gx#self-hosted-deployment-pattern) to run the GX Agent with Docker, connect the GX Agent to your target Data Sources, and use the GX Cloud web UI to define your Data Assets, create Expectations, and run Validations. When you've finished testing GX Cloud, you move to an [org-hosted deployment](./about_gx.md#org-hosted-deployment-pattern) and deploy the GX Agent to your organization's development, staging, or production cloud services environment. The GX Agent serves all GX Cloud users within your organization. It can be run as part of your development or production workflows. To learn more about the GX Agent and deployment patterns, see [About GX Cloud](./about_gx.md).
```
Comment (annabarr) [inline docs/docusaurus/docs/cloud/deploy_gx_agent.md]: The term "try GX Cloud" has been confusing to me in the past– I don't associate "try" with a temporary or test setup, I associate it with getting started. 

If we want to emphasize the ephemeral local setup, maybe we say "To test GX Cloud, you may use a self-hosted deployment...."? What do you think @kwcanuck ?

Comment (kwcanuck) [reply docs/docusaurus/docs/cloud/deploy_gx_agent.md]: @annabarr _test_ works for me. I've made the necessary edits.

File: docs/docusaurus/versioned_docs/version-0.18/cloud/connect/connect_python.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ Learn how to use GX Cloud from a Python script or interpreter, such as a Jupyter
 
 To get the most out of GX Cloud, GX recommends deploying the GX Agent. If you do not deploy the GX Agent, some features and functionality might be unavailable. To deploy the GX Agent, see [Deploy the GX Agent](../deploy_gx_agent.md).
 
+If you don't want to deploy the GX Agent, use the GX API to create a Data Source for your GX Cloud organization. See [Connect to a Data Source](/oss/guides/connecting_to_your_data/connect_to_data_lp.md).
+
```
Comment (Shinnnyshinshin) [inline docs/docusaurus/versioned_docs/version-0.18/cloud/connect/connect_python.md:19]: Love this clarity. Thanks

File: docs/docusaurus/docs/cloud/deploy_gx_agent.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ import Tabs from '@theme/Tabs';
...
 
-The GX Agent is typically deployed in your organization's deployment environment, for example, in a development, staging, or production cloud services environment. The GX Agent serves all GX Cloud users within your organization. It can be run as part of your development or production workflows. To learn more about the GX Agent and deployment patterns, see [About GX Cloud](./about_gx.md).
+To test GX Cloud, you use a [self-hosted deployment](./about_gx#self-hosted-deployment-pattern) to run the GX Agent with Docker, connect the GX Agent to your target Data Sources, and use the GX Cloud web UI to define your Data Assets, create Expectations, and run Validations. When you've finished testing GX Cloud, you move to an [org-hosted deployment](./about_gx.md#org-hosted-deployment-pattern) and deploy the GX Agent to your organization's development, staging, or production cloud services environment. The GX Agent serves all GX Cloud users within your organization. It can be run as part of your development or production workflows. To learn more about the GX Agent and deployment patterns, see [About GX Cloud](./about_gx.md).
```
Comment (Shinnnyshinshin) [inline docs/docusaurus/docs/cloud/deploy_gx_agent.md:15]: 🚀

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you so much for this update @kwcanuck LGTM

**Review by annabarr — APPROVED** [review]
lgtm!

**Pull request merged**